src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs

```diff
@@ -793,6 +793,65 @@ void ConfigureListenOptions(ListenOptions listenOptions)
             await AssertConnectionResult(stream, true, expectedBody);
         }
 
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.MacOSX, SkipReason = "ALPN not supported")]
+        public async Task ServerOptionsSelectionCallback_SetsALPN()
+        {
+            static void ConfigureListenOptions(ListenOptions listenOptions)
+            {
+                listenOptions.UseHttps((_, _, _, _) =>
+                    ValueTask.FromResult(new SslServerAuthenticationOptions()
+                    {
+                        ServerCertificate = _x509Certificate2,
```

Comment:
Can we add a test case confirming that adding

```suggestion
                        ServerCertificate = _x509Certificate2,
                        ApplicationProtocols = new()
```

disable ALPN? Assuming it does ofc.

Reply:
It worked on Windows but fails on Ubuntu 😢. I'll file a runtime bug for that.

Reply:
Fixed in rc1 (main), https://github.com/dotnet/runtime/pull/55772. Waiting for a runtime update.